src/librbd/internal.cc

```diff
@@ -531,7 +531,10 @@ int remove_object_map(ImageCtx *ictx) {
       }
 
       r = ictx->image_watcher->notify_snap_create(snap_name);
-      if (r == -EEXIST) {
+      if (r == 0) {
+        notify_change(ictx->md_ctx, ictx->header_oid, ictx);
+        return 0;
+      } else if (r == -EEXIST) {
         return 0;
```

Comment:
will the notification still happen if the notify times out, but the lock owner successfully created the snapshot (the EEXIST case here?) It seems more fragile to separate the notify_change() from the work being done.

If the race is just for the snap_create() initiator, can we just increment refresh_seq here instead of moving the notification away from the lock owner?